src/index.ts

```diff
@@ -100,9 +108,30 @@ function ViteImageOptimizer(optionsParam: Options = {}): Plugin {
 
   const processFile = async (filePath: string, buffer: Buffer) => {
     try {
-      const engine: Function = /\.svg$/.test(filePath) ? applySVGO : applySharp;
-      const newBuffer: Buffer = await engine(filePath, buffer);
+      let newBuffer: Buffer;
+
+      let isCached: boolean;
+      const cachedFilePath = join(options.cacheLocation, filePath);
+      if (options.cache === true && fs.existsSync(cachedFilePath)) {
```

Comment:
Since we are checking if a file is cached based on its path existing in the default cache folder, let's consider a scenario:

Consider the following images:
- `1.png` located at `src/assets/1.png`
- `2.png` located at `src/assets/2.png`
- `3.png` located at `src/assets/3.png`

Now, let's walk through the process:
1. We build the app -> Vite plugin runs -> Files don't exist in cache -> Engine runs -> Files are stored back into the cache.
2. We build our app again -> Vite plugin runs -> Files already exist in cache -> Engine is skipped -> File buffers are restored.

Now, let's introduce a situation:
The developer deletes the file `1.png` at `src/assets/1.png` and replaces it with a different file, but with the same name `1.png`.

Now, we build the app again:
Vite plugin runs -> Files already exist in cache (We don't know that the content of the file `1.png` has changed) -> Engine is skipped -> File buffers are restored.

As a result, we will end up with an earlier version of `1.png` (version before the developer deleted and renamed the file)

I think just checking the file path's existence isn't enough here. I'd like to know your opinion on this. Tools like `prettier` and `eslint` handle this by using a metadata file with hashed details of the files. I'm not sure about their exact implementation, though.

Reply:
In our setup, we found that paths are always suffixed with a hash (generated by Vite). This means your scenario would become:

* `1.png` is processed with filename `1-abc1234.png` (Vite appended the content hash)
* `1-abc1234.png` is stored in the cache
* A new version of `1.png` is created, and now passed in as `1-def5678.png`. This does not exist yet in the cache, so is recreated.

I'm not sure this is always the behaviour for Vite ([though it looks like it is](https://github.com/vitejs/vite/blob/d36d6fb91d50b338f689e6c554e3896b3d739390/docs/guide/assets.md?plain=1#L25), and [here's the hashing function](https://github.com/vitejs/vite/blob/d36d6fb91d50b338f689e6c554e3896b3d739390/packages/vite/src/node/utils.ts#L983) referenced [in the asset handling function](https://github.com/vitejs/vite/blob/d36d6fb91d50b338f689e6c554e3896b3d739390/packages/vite/src/node/plugins/asset.ts#L305)). With that handled by Vite, do you think a metadata file is still needed?

Reply:
This is fantastic, Maarten! Thank you for providing the hashing function. It appears that the hash is derived from the file's buffer, which works in our favour. I don't believe the metadata file will be necessary since it might only introduce extra overhead. Lets go with this!